### PR TITLE
Refactoring

### PR DESCRIPTION
**Pull request by kamilsa**
This is what is done so far:
 
Remove unused include
1. Apply clang-format
2. more clearly value name
3. Correct my wrong English
4. Remove magic numbers and hard coded values.
5. Gather if (getenv("IROHA_HOME") == nullptr) {...} into config. (done by introducing config::get_iroha_home())
6. Mark Deprecated to JVM and smart contracts (review if it is done properly)
7. Replace assertions with exceptions except in test codes (none of that found)

### CODE REVIEW

File: core/infra/virtual_machine/jvm/java_virtual_machine.cpp [hunk 1]
```diff
@@ -40,7 +39,7 @@ namespace jvm {
     std::unique_ptr<JavaContext> initializeVM(const std::string& packageNameUnderInstances, const std::string& contractName) {
 
         const auto IrohaHome = []() {
-            const auto p = getenv("IROHA_HOME");
+            const auto p = config::get_iroha_home();
             if (p == nullptr) {
                 throw "[FATAL] Set environment $IROHA_HOME";
```
Comment (motxx) [inline core/infra/virtual_machine/jvm/java_virtual_machine.cpp]: 1. This shows:
```
terminate called after throwing an instance of 'char const*'
Aborted
```
C++ cannot throw message directory. It needs some exception call. (Sorry, actually this is my mistake ;(

2. I think it is better to include this nullptr check into `config::get_iroha_home()`. If this change is applied, all variables such as `IrohaHome` can be removed and replaced with `config::get_iroha_home()`, and in that case, exception message `"[FATAL] Set environment $IROHA_HOME"` is better.

Comment (motxx) [inline core/infra/virtual_machine/jvm/java_virtual_machine.cpp]: 1. This shows:
```
terminate called after throwing an instance of 'char const*'
Aborted
```
`throw` needs an exception class in order to show message correctly. (actually, this is my mistake. sorry ;(

2. I think it is better to include this nullptr check into `config::get_iroha_home()`. If this change is applied, all variables such as `IrohaHome` can be removed and replaced with `config::get_iroha_home()`, and in that case, exception message `"[FATAL] Set environment $IROHA_HOME"` is better.

**Comment by motxx** [comment]
In my quick review, fixes of `6. Mark Deprecated to JVM and smart contracts` looks good to me.

**Comment by luckychess** [comment]
Looks good for me.

**Pull request merged**